### PR TITLE
Add google_network_connectivity_policy_based_route resource

### PR DESCRIPTION
**Pull request by rileykarson**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/15414, b/249847932

I didn't test the `interconnect_attachment` target, given we don't own an interconnect attachment ourselves to test against.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_network_connectivity_policy_based_route`
```

### CODE REVIEW

File: mmv1/products/networkconnectivity/PolicyBasedRoute.yaml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    output: true
+    description: |
+      Time when the policy-based route was created.
+  - !ruby/object:Api::Type::Time
+    name: 'kind'
+    output: true
+    description: |
+      Type of this resource.
```
Comment (shuyama1) [inline mmv1/products/networkconnectivity/PolicyBasedRoute.yaml:171]: Do we need these fields in Terraform? especially `updateTime` and `kind`. Non-blockers though.

File: mmv1/products/networkconnectivity/PolicyBasedRoute.yaml [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        description: |
+          The source IP range of outgoing packets that this policy-based route applies to. Default is "0.0.0.0/0" if protocol version is IPv4.
+      - !ruby/object:Api::Type::String
+        name: 'destRange'
+        default_value: "0.0.0.0/0"
+        description: |
+          The destination IP range of outgoing packets that this policy-based route applies to. Default is "0.0.0.0/0" if protocol version is IPv4.
+  - !ruby/object:Api::Type::Enum
+    name: 'nextHopOtherRoutes'
+    exactly_one_of:  # next_hop union
```
Comment (shuyama1) [inline mmv1/products/networkconnectivity/PolicyBasedRoute.yaml:111]: Do we need `exactly_one_of` here or `conflicts`? Like is the `next_hop` union field required?

File: mmv1/products/networkconnectivity/PolicyBasedRoute.yaml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    description: |
+      The IP address of a global-access-enabled L4 ILB that is the next hop for matching packets.
+  - !ruby/object:Api::Type::Integer
+    name: 'priority'
+    default_value: 1000
+    description: |
+      The priority of this policy-based route. Priority is used to break ties in cases where there are more than one matching policy-based routes found. In cases where multiple policy-based routes are matched, the one with the lowest-numbered priority value wins. The default value is 1000. The priority value must be from 1 to 65535, inclusive.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'virtualMachine'
+    exactly_one_of:  # target union
```
Comment (shuyama1) [inline mmv1/products/networkconnectivity/PolicyBasedRoute.yaml:132]: same question for the `target` union field.

**Review by shuyama1 — COMMENTED** [review]
overall looks good. Only small comments

File: mmv1/products/networkconnectivity/PolicyBasedRoute.yaml (hunk 3 above)
Comment (rileykarson) [reply mmv1/products/networkconnectivity/PolicyBasedRoute.yaml:132]: No, good catch! One of my tests is gonna fail :(

I added this union after writing the basic test so hadn't run it locally

File: mmv1/products/networkconnectivity/PolicyBasedRoute.yaml (hunk 2 above)
Comment (rileykarson) [reply mmv1/products/networkconnectivity/PolicyBasedRoute.yaml:111]: Yeah, this one if required- oneofs are a bit weird and can't be made required in the API schema, but can be checked at runtime. Configs w/o either fail.

File: mmv1/products/networkconnectivity/PolicyBasedRoute.yaml (hunk 1 above)
Comment (rileykarson) [reply mmv1/products/networkconnectivity/PolicyBasedRoute.yaml:171]: Nope, we don't need them. I just erred on the side of including them.

Comment (shuyama1) [reply mmv1/products/networkconnectivity/PolicyBasedRoute.yaml:171]: Makes sense. Including them is fine! I just want to confirm personally as I sometimes ask contributors to exclude these and want to make sure I didn't ask them to remove something that end up needed

**Review by shuyama1 — APPROVED** [review]
assuming tests pass

**Review by shuyama1 — APPROVED** [review]

**Pull request merged**